docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+# Context Menu Manager (API)
+
+### Table of content
+- [Introduction](#Introduction)
+    - [Context menu UI behavior](#Context menu UI behavior)
+    - [Context-Menu config structure](#Context Menu config structure)
+    - [Structure tree](#Structure tree)
+- [API methods]()
+  - [Create a group](#Create a group)
+  - [Update a group](#Update a group)
+  - [Delete a group](#Delete a group)
+  - [Add item](#Add item)
+  - [Update item](#Update item)
+  - [Delete item](#Delete item)
+  - [Get config](#Get config)
+- [Understanding the usage flow]()
+  - [Registration of new items](#Registration of new items)
+  - [Triggering the context menu flow](#Triggering the context menu flow)
+- [Example of usage]()
+
+## Introduction
+The context menu manager provides a convenient way for managing the context menu structure.
+The manager is responsible for building a tree of nested groups/items that forms the actual structure of the context menu, as presented in the UI.   
+The manager is based on CRUD principles while providing simple functions for adding, updating, deleting, and in the end - provides a function for reading the main tree that have been created.
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
```

Comment:
Isn't it more of a UI thing? The manager doesn't care about it

Reply:
No. It's important to understand how to use the API.

Reply:
I'm not sure.
@matipojo WDYT?